src/main/java/ch/njol/skript/sections/EffSecSendResourcePack.java

```diff
@@ -0,0 +1,214 @@
+			"\t\tset the resource pack uuid to \"1\"",
+			"\t\tset the resource pack hash to \"Hash\"",
+			"\t\tset the resource pack prompt to \"Please Download\"",
+			"\t\tforce the player to accept"
+})
+@Since("2.4, INSERT VERSION (section)")
+public class EffSecSendResourcePack extends EffectSection {
+
+	public static class ResourcePackEvent extends Event {
+		@Nullable
+		private String id, hash, prompt;
+		@Nullable
+		private Boolean force;
+
+		public ResourcePackEvent() {}
+
+		public void setId(String id) {
+			this.id = id;
+		}
+		public void setHash(String hash) {
+			this.hash = hash;
+		}
+		public void setPrompt(String prompt) {
+			this.prompt = prompt;
+		}
+		public void setForce(Boolean force) {
+			this.force = force;
+		}
+
+		public String getId() {
+			return id;
+		}
+		public String getHash() {
+			return hash;
+		}
+		public String getPrompt() {
+			return prompt;
+		}
+		public Boolean getForce() {
+			return force;
+		}
+
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+	}
+
+	static {
+		Skript.registerSection(EffSecSendResourcePack.class,
+			"send [a|the] resource pack [at|from [[the] URL]] %string% to %players% [using|with]",
+			"send [a|the] resource pack [at|from [[the] URL]] %string% with hash %string% to %players%"
+			);
+
+	}
+
+	@UnknownNullability
+	@SuppressWarnings("null")
+	private Expression<String> url;
+	@UnknownNullability
+	@SuppressWarnings("null")
+	private Expression<Player> recipients;
+	@Nullable
+	private Trigger trigger;
+	@Nullable
+	private Expression<String> oldHash;
+	private int pattern;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult, @Nullable SectionNode sectionNode, List<TriggerItem> triggerItems) {
+		this.url = (Expression<String>) exprs[0];
+		this.pattern = matchedPattern;
+		if (matchedPattern == 2) {
+			this.oldHash = (Expression<String>) exprs[1];
+			this.recipients = (Expression<Player>) exprs[2];
+		} else {
+			this.recipients = (Expression<Player>) exprs[1];
+		}
+
+		if (sectionNode != null) {
+			AtomicBoolean delayed = new AtomicBoolean(false);
+			Runnable afterLoading = () -> delayed.set(!getParser().getHasDelayBefore().isFalse());
+			trigger = loadCode(sectionNode, "resource", afterLoading, ResourcePackEvent.class);
+			if (delayed.get()) {
+				Skript.error("Delays can't be used within a Resource Pack Effect Section");
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+
+	@Override
+	protected @Nullable TriggerItem walk(Event event) {
+		debug(event, true);
+		assert url != null;
+		String address = url.getSingle(event);
+		UUID uuid = null;
+		String hash =  this.oldHash == null ? "defaultHash" : this.oldHash.getSingle(event);
+		String prompt = "The server has requested you to download a resource pack";
+		boolean force = false;
```

Comment:
We shouldn't be using default values if not supplied, we should simply use the method that corresponds to what values we did get, or we should pass nulls.